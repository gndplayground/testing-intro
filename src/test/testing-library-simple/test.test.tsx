import React from "react";
import { render, screen } from "@testing-library/react";

import Simple from "./Simple";

// The test cases here are for demo purpose.

describe("simple", () => {
  it("Should render", () => {
    const { container } = render(<Simple />);

    expect(container.firstChild).toMatchSnapshot();

    screen.debug();
  });

  it("Should render with name", () => {
    const { container, rerender } = render(<Simple />);

    expect(container.firstChild).toMatchSnapshot();

    rerender(<Simple name="Tom" />);

    expect(container.firstChild).toMatchSnapshot();

    screen.debug();
  });

  it("Should render correct text", () => {
    const { getByText } = render(<Simple name="Jean" />);

    expect(getByText("Hello, Jean!"));
  });
});
