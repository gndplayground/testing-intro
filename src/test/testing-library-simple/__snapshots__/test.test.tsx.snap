// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple Should render 1`] = `
<section>
  <p>
    Hello
  </p>
</section>
`;

exports[`simple Should render with name 1`] = `
<section>
  <p>
    Hello
  </p>
</section>
`;

exports[`simple Should render with name 2`] = `
<section>
  <p>
    Hello
    , Tom!
  </p>
</section>
`;
